tests/phpunit/tests/dependencies/scripts.php

```diff
@@ -24,10 +33,12 @@ class Tests_Dependencies_Scripts extends WP_UnitTestCase {
 	public function set_up() {
 		parent::set_up();
 		$this->old_wp_scripts = isset( $GLOBALS['wp_scripts'] ) ? $GLOBALS['wp_scripts'] : null;
+		$this->old_wp_styles  = isset( $GLOBALS['wp_styles'] ) ? $GLOBALS['wp_styles'] : null;
 		remove_action( 'wp_default_scripts', 'wp_default_scripts' );
 		remove_action( 'wp_default_scripts', 'wp_default_packages' );
 		$GLOBALS['wp_scripts']                  = new WP_Scripts();
 		$GLOBALS['wp_scripts']->default_version = get_bloginfo( 'version' );
+		$GLOBALS['wp_styles']                   = new WP_Styles();
```

Comment:
Are we actually touching the global `wp_styles` object at any point during these tests? Curious why it needs to be reset?

Reply:
Yes, because the tests call `wp_print_footer_scripts()` which in turn calls `_wp_footer_scripts()` and then `print_late_styles()`. I found that another test somewhere else in the test suite outside the `scripts` group was populating the `$wp_styles` global with enqueued styles, and this caused unexpected styles to get printed. This broke my comparison between just having expected scripts, since styles were then added as well.